cmd/commands/git-source.go

```diff
@@ -0,0 +1,204 @@
+	var (
+		insCloneOpts *git.CloneOptions
+		gsCloneOpts  *git.CloneOptions
+	)
+
+	cmd := &cobra.Command{
+		Use:   "create runtime_name git-source_name git-src-repo_full_path",
+		Short: "add a new git-source to an existing runtime",
+		Example: util.Doc(`
+			<BIN> git-source create runtime_name git-source-name https://github.com/owner/repo-name/my-workflow
+		`),
+		PreRun: func(cmd *cobra.Command, args []string) {
+			ctx := cmd.Context()
+
+			if len(args) < 1 {
+				log.G(ctx).Fatal("must enter runtime name")
+			}
+
+			if len(args) < 2 {
+				log.G(ctx).Fatal("must enter git-source name")
+			}
+
+			if gsCloneOpts.Repo == "" {
+				log.G(ctx).Fatal("must enter a valid value to --git-src-repo. Example: https://github.com/owner/repo-name/path/to/workflow")
+			}
+
+			if gsCloneOpts.Auth.Password == "" {
+				gsCloneOpts.Auth.Password = insCloneOpts.Auth.Password
+			}
+
+			insCloneOpts.Parse()
+			gsCloneOpts.Parse()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+
+			err := RunCreateGitSource(ctx, &GitSourceCreateOptions{
+				insCloneOpts: insCloneOpts,
+				gsCloneOpts:  gsCloneOpts,
+				gsName:       args[1],
+				runtimeName:  args[0],
+				fullGsPath:   gsCloneOpts.Path(),
+			})
+
+			if err != nil {
+				return fmt.Errorf("failed to create git-source: %w", err)
+			}
+
+			return nil
+		},
+	}
+
+	insCloneOpts = git.AddFlags(cmd, &git.AddFlagsOptions{
+		CreateIfNotExist: true,
+		FS:               memfs.New(),
+	})
+	gsCloneOpts = git.AddFlags(cmd, &git.AddFlagsOptions{
+		Prefix:           "git-src",
+		Optional:         true,
+		CreateIfNotExist: true,
+		FS:               memfs.New(),
+	})
+
+	return cmd
+}
+
+func RunCreateGitSource(ctx context.Context, opts *GitSourceCreateOptions) error {
+	gsRepo, gsFs, err := opts.gsCloneOpts.GetRepo(ctx)
+	if err != nil {
+		return err
+	}
+
+	fi, err := gsFs.ReadDir(".")
+
+	if err != nil {
+		return fmt.Errorf("failed to read files in git-source repo. Err: %w", err)
+	}
+
+	if len(fi) == 0 {
+		if err = createDemoWorkflowTemplate(gsFs, opts.gsName, opts.runtimeName); err != nil {
+			return fmt.Errorf("failed to create demo workflowTemplate: %w", err)
+		}
+
+		_, err = gsRepo.Persist(ctx, &git.PushOptions{
+			CommitMsg: fmt.Sprintf("Created demo workflow template in %s Directory", opts.gsCloneOpts.Path()),
+		})
+
+		if err != nil {
+			return fmt.Errorf("failed to push changes. Err: %w", err)
+		}
+	}
+
+	appDef := &runtime.AppDef{
+		Name: opts.gsName,
+		Type: application.AppTypeDirectory,
+		URL:  opts.gsCloneOpts.Repo,
+	}
+	if err := appDef.CreateApp(ctx, nil, opts.insCloneOpts, opts.runtimeName, store.Get().CFGitSourceType, nil); err != nil {
+		return err
+	}
+
+	log.G(ctx).Infof("done creating a new git-source: '%s'", opts.gsName)
+
+	return nil
+}
+
+func createDemoWorkflowTemplate(gsFs fs.FS, gsName, runtimeName string) error {
+	var err error
+
+	gsPath := gsFs.Join(apstore.Default.AppsDir, gsName, runtimeName)
```

Comment:
the gsPath can contain the "demo-wf-template.yaml" string here, and then there is no need to call `gsFs.join` again in line 199